slave_monitor/js/ClusterRunnerSlaveMonitor.js

```diff
@@ -108,17 +108,20 @@ cls._startVisualization = function(visualizers)
 
         var nodes = [], links = [];
         var graphStateChanged = false;
-        visualizers.map(function (visualizer) {
+        visualizers.forEach(function (visualizer) {
             graphStateChanged = visualizer.update() || graphStateChanged;
             nodes = nodes.concat(visualizer.getNodes());
             links = links.concat(visualizer.getLinks());
         });
+        d3.selectAll('.slaveCircle').on('click', function(e, i) {
+            window.open('http://' + e.slaveDatum.url + '/v1', '');
+        });
```

Comment:
I think it should be fine. The documentation says "If an event listener was already registered for the same type on the selected element, the existing listener is removed before the new listener is added."

https://github.com/mbostock/d3/wiki/Selections#on